### PR TITLE
fix(trade): display Price updated banner in Confirmation modal

### LINKED ISSUE #2639: Deploy smart-contracts to all networks
Currently, TWAP is deployed only on Goerli, need to support all networks.

## TODO
 - https://github.com/cowprotocol/cowswap/blob/426668c9e405628753ead82cb67ce4429be168af/src/modules/twap/const.ts#L30

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #2639

Before:
image

Now:
image

image

  # To Test (Swap/Limit/Advanced)

1. Open a confirmation order
2. Wait until price is changed
- [ ] The `Price updated` banner should be shown
- [ ] The confirmation button should be disabled
2. Click on "Accept"
- [ ] The `Price updated` banner should be hidden
- [ ] The confirmation button should be enabled

### CODE REVIEW

**Review by elena-zh — CHANGES_REQUESTED** [review]
Hey @shoom3301 , nice!
However: 
1. Swap modal: the warning does not look good there. Besides, it would be great to have a hover/click effect on the Accept button 
hover
Screenshot_7
Screenshot_8
2. Limit orders: the warning does not appear there. I noticed, that amounts are staying stale on the modal and on the form. The only 1 thing I see when quote/price is updated, is that price deviation indicator is changed, and a fee  on the Limit orders form. So, I assume, there is no need in this warning on the Limit orders modal, correct?
updated

3. Advanced orders: the warning appear each time when quote request is running. But it should not appear until a price/amount is changed. 
nothing changed

**Comment by shoom3301** [comment]
1. Fixed, thank you!
2. Yes, you are right. The banner should never be display in Limit orders, because the price here is static
3. Actually, the amount changed, you don't see it because we trim the displayed amount. Please, look at the full amount (hover on the amount)

image

**Comment by elena-zh** [comment]
Hey @shoom3301 , thanks for the explanation. 
Case 1 is fixed. 
Case 3 is also understandable to me. 

However, I see a difference how it works on the Swap page in comparison with the Prod: 
1. On the Swap page  in the PR To field remains unchanged. Price updated banner appears when Receive (incl.fee) is changed
image

2. On Prod the amount in the TO field is changed --> banner appear. 

Could you please take a look at it?

**Review by elena-zh — APPROVED** [review]

File: src/common/pure/InlineBanner/index.tsx [hunk 1]
```diff
@@ -57,6 +57,7 @@ const Wrapper = styled.span<{ color: string }>`
     width: var(--size);
     height: var(--size);
     object-fit: contain;
+    stroke: none !important;
```
Comment (anxolin) [inline src/common/pure/InlineBanner/index.tsx:60]: I will tell @fairlighteth  :P

File: src/modules/trade/pure/TradeConfirmation/index.tsx [hunk 2]
```diff
@@ -60,10 +69,37 @@ export function TradeConfirmation(props: TradeConfirmationProps) {
           priceImpactParams={priceImpact}
         />
         {children}
-        <ButtonPrimary onClick={onConfirm} disabled={isConfirmDisabled} buttonSize={ButtonSize.BIG}>
+        {isPriceChanged && <PriceUpdatedBanner onClick={resetPriceChanged} />}
```
Comment (anxolin) [inline src/modules/trade/pure/TradeConfirmation/index.tsx:72]: Great you use the oportunity to refactor

**Review by anxolin — APPROVED** [review]

File: src/common/pure/InlineBanner/index.tsx (hunk 1 above)
Comment (shoom3301) [reply src/common/pure/InlineBanner/index.tsx:60]: It's because of legacy styles in Swap confirm modal...

**Pull request merged**